components/switch/fake/switch.go

```diff
@@ -0,0 +1,108 @@
+// Package fake implements fake switches with different position counts.
+package fake
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	switch_component "go.viam.com/rdk/components/switch"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/resource"
+)
+
+var (
+	model2Way  = resource.DefaultModelFamily.WithModel("fake-2way")
+	model3Way  = resource.DefaultModelFamily.WithModel("fake-3way")
+	model10Way = resource.DefaultModelFamily.WithModel("fake-10way")
+)
+
+// Config is the config for a fake switch.
+type Config struct {
+	resource.TriviallyValidateConfig
+}
+
+func init() {
+	// Register all three switch models
+	resource.RegisterComponent(switch_component.API, model2Way, resource.Registration[switch_component.Switch, *Config]{
+		Constructor: func(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger logging.Logger) (switch_component.Switch, error) {
+			return NewSwitch(ctx, deps, conf, logger, 2)
+		},
+	})
+	resource.RegisterComponent(switch_component.API, model3Way, resource.Registration[switch_component.Switch, *Config]{
+		Constructor: func(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger logging.Logger) (switch_component.Switch, error) {
+			return NewSwitch(ctx, deps, conf, logger, 3)
+		},
+	})
+	resource.RegisterComponent(switch_component.API, model10Way, resource.Registration[switch_component.Switch, *Config]{
+		Constructor: func(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger logging.Logger) (switch_component.Switch, error) {
+			return NewSwitch(ctx, deps, conf, logger, 10)
+		},
+	})
+}
+
+// Switch is a fake switch that can be set to different positions.
+type Switch struct {
+	resource.Named
+	resource.TriviallyCloseable
+	mu            sync.Mutex
+	logger        logging.Logger
+	position      uint32
+	positionCount int
+}
+
+// NewSwitch instantiates a new switch of the fake model type.
+func NewSwitch(
+	ctx context.Context,
+	deps resource.Dependencies,
+	conf resource.Config,
+	logger logging.Logger,
+	positionCount int,
+) (switch_component.Switch, error) {
+	s := &Switch{
+		Named:         conf.ResourceName().AsNamed(),
+		logger:        logger,
+		position:      0,
+		positionCount: positionCount,
+	}
+	if err := s.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+	return s, nil
+}
+
+// Reconfigure reconfigures the switch atomically and in place.
+func (s *Switch) Reconfigure(_ context.Context, _ resource.Dependencies, conf resource.Config) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	return nil
+}
```

Comment:
```suggestion
```
Don't need if you add `resource.AlwaysRebuild` above. This is doing nothing anyway.